### PR TITLE
Fix - update to Net::AMQP::RabbitMQ 0.31, where delivery_tag is UInt64

### PR DESCRIPTION
**Pull request by eilara**
From Net::AMQP::RabbitMQ Changes:

0.100000 - 2015-03-09
- Support for 64-bit delivery tags via Math::Int64 (manchicken)

But Test::Net::RabbitMQ stores delivery tags as ints, which makes it incompatible with the implementation it is mocking.

### CODE REVIEW

File: lib/Test/Net/RabbitMQ.pm [hunk 1]
```diff
@@ -156,17 +158,27 @@ has queues => (
...
+    my $self = shift;
+    $self->_set_delivery_tag( $self->delivery_tag + 1 );
+}
+
+sub _dec_delivery_tag {
+    my $self = shift;
+    $self->_set_delivery_tag( $self->delivery_tag - 1 );
+}
+
+sub _reset_delivery_tag {
```
Comment (autarch) [inline lib/Test/Net/RabbitMQ.pm]: Just use a `clearer` on the attribute. Making an explicit sub is unnecessary.

File: lib/Test/Net/RabbitMQ.pm [hunk 2]
```diff
@@ -156,17 +158,23 @@ has queues => (
...
-    handles => {
-        _inc_delivery_tag   => 'inc',
-        _dec_delivery_tag   => 'dec',
-        _reset_delivery_tag => 'reset',
-    },
+    is      => 'rw',
+    isa     => 'Math::UInt64',
+    default => sub { uint64(0) },
+    clearer => '_reset_delivery_tag',
+    writer  => '_set_delivery_tag',
```
Comment (autarch) [inline lib/Test/Net/RabbitMQ.pm:165]: Why does this need a writer?

Comment (eilara) [inline lib/Test/Net/RabbitMQ.pm:165]: Is $self->{delivery_tag}++ better than using reader/writer in _inc_delivery_tag?

I admit this is a degenerate case, and probably the worst possible pedagogical example of the usefulness of this encapsulation, but there could be triggers, overrides, etc.

Comment (autarch) [inline lib/Test/Net/RabbitMQ.pm:165]: You were right to have a writer. I had missed the need for it elsewhere in the code. I ended up merging this as-is.

**Pull request merged**